tests/test_base_task.py

```diff
@@ -143,3 +149,92 @@ def test_quickbase_task_run_upsert_and_json_receipt_output_target_success(
 
 def test_base_pipeline_name(task_pipeline_animals):
     assert task_pipeline_animals.pipeline_name == "Animals"
+
+
+def test_base_sql_task_missing_sql_query_or_sql_file_error(pipeline_name):
+    class MissingRequiredPropertiesQueryTask(SQLQueryExtractTask):
+        pass
+
+    task = MissingRequiredPropertiesQueryTask(pipeline=pipeline_name, stage="Extract")
+    with pytest.raises(
+        AttributeError, match="Property sql_query or sql_file must be set."
+    ):
+        task.get_dataframe()
+
+
+def test_base_sql_task_custom_dwclient(task_sql_extract_animal_names):
+    dwclient = task_sql_extract_animal_names.dwclient
+    assert isinstance(dwclient, DWClient)
+    dwclient.init_engine()
+    assert dwclient.engine.name == "sqlite"
+
+
+def test_base_sql_task_sql_query(task_sql_extract_animal_names):
+    assert (
+        task_sql_extract_animal_names.sql_query
+        == """
+        select animal_id, name from animal_name
+        """
+    )
+
+
+def test_base_sql_task_sql_file(task_sql_extract_animal_colors):
+    assert (
+        task_sql_extract_animal_colors.sql_file
+        == "tests/fixtures/sql/animal_color_query.sql"
+    )
+
+
+def test_base_sql_task_sql_query_get_dataframe_return_dataframe(
+    task_sql_extract_animal_names,
+):
+    df = task_sql_extract_animal_names.get_dataframe()
+    assert isinstance(df, pd.DataFrame)
+
+
+def test_base_sql_task_sql_file_get_dataframe_return_dataframe(
+    task_sql_extract_animal_colors,
+):
+    df = task_sql_extract_animal_colors.get_dataframe()
+    assert isinstance(df, pd.DataFrame)
+
+
+def test_base_sql_task_sql_query_parameters_used(
+    pipeline_name, sqlite_dwclient, task_sql_extract_animal_colors
+):
+    foo_val, bar_val = 42, "apple"
+
+    class SQLQueryWithParameters(SQLQueryExtractTask):
+        stage = luigi.Parameter("Extract")
+
+        @property
+        def dwclient(self) -> DWClient:
+            return sqlite_dwclient
+
+        @property
+        def sql_query(self) -> str:
+            return """
+            select
+                :foo_val as foo,
+                :bar_val as bar
+            """
+
+        @property
+        def sql_query_parameters(self) -> dict:
+            return {"foo_val": foo_val, "bar_val": bar_val}
```

Comment:
Could this be shifted to `tests/fixtures/tasks/extract.py` or `conftest.py`? This is large enough that defining it in the test feels messy

Reply:
Yep, good call, don't see any reason why it can't live in `tests/fixtures/tasks/extract.py`.